content/blog/use-redfish-and-ipv6-to-find-lost-servers.md

```diff
@@ -0,0 +1,475 @@
+
+If you have the list of lost MAC addresses and you know that
+<a href="https://datatracker.ietf.org/doc/html/rfc7217" target="_blank">RFC 7217</a>
+is set to disabled, you just have to compute the corresponding IPv6 link-local addresses with the `ipv6calc`
+tool and verify their accuracy.
+
+> NOTE: RFC 7217 is disabled by default as explained [above](#ipv6-link-local-addresses).
+
+The following example computes and stores IPv6 link local addresses
+in a variable. Then it tests whether they correspond to a
+Redfish conformant (iLO) address by requesting the property `RedfishVersion` from the
+`ServiceRoot` URI.
+
+```shell
+MacList="5C:ED:8C:01:0D:7C 5C:ED:8C:6B:06:17"
+LL_LIST="$(for mac in $MacList ;
+              do echo -n "fe80::" ; ipv6calc --quiet --in mac $mac --out eui64
+           done
+         )"
+
+for ll in $LL_LIST ; do
+    curl --noproxy \* --insecure --silent https://[${ll}%eno1]/redfish/v1 | \
+         jq '.RedfishVersion'
+done
+"1.20.0"
+"1.13.0"
+```
+
+You can use iLOrest to verify that the computed list of link local addresses
+corresponds to iLO addresses.
+However, you will need to provide credentials, even for accessing the Redfish root URI.
+
+```shell
+u="ilo-user"
+p="ilo-password"
+
+for ll in $LL_LIST ; do
+    ilorest --nologo login [${ll}%eno1] -u $u -p $p >/dev/null
+    ilorest --nologo get RedfishVersion --select ServiceRoot
+    ilorest --nologo logout >/dev/null
+done
+RedfishVersion=1.20.0
+RedfishVersion=1.13.0
+```
+
+### Configuration recovery
+
+The last step to completely recover from the lost iLOs is to re-configure their
+network parameters. From the brick-saver, the following example
+shows a generic PATCH request and its associated payload
+to configure a minimal static IPv4 configuration of a lost iLO.
+
+With the previous examples, it should be easy for you to adapt it
+for iLOrest, cURL or PowerShell. Other examples for configuring iLO network
+parameters are presented in the
+<a href="https://servermanagementportal.ext.hpe.com/docs/redfishclients/ilorest-userguide/examplecommandsscripts/#configure-ilo-ip-addresses"
+target="_blank">documentation</a>.
+
+```text
+PATCH request URL
+https://[FE80::9618:82FF:FE71:A07B%eno1]/redfish/v1/Managers/1/EthernetInterfaces/1/
+
+Payload:
+
+{
+    "DHCPv4": {
+        "DHCPEnabled": false,
+        "UseDNSServers": false,
+        "UseDomainName": false,
+        "UseGateway": false,
+        "UseNTPServers": false,
+        "UseStaticRoutes": false
+    },
+    "IPv4StaticAddresses": [
+        {
+            "Address": "192.168.1.44",
+            "Gateway": "192.168.1.1",
+            "SubnetMask": "255.255.252.0"
+        }
+    ]
+}
+```
+
+## Conclusion
+
+This article presented a method with simple PowerShell and Linux commands
+to find back lost servers after a
+move from one network to another without any configuration preparation prior to the move.
+
+As this method uses IPv6, many of its concepts have been introduced,
```

Comment:
Should it say "including the its autoconfiguration process" or should it say "including its autoconfiguration process"?

In the second sentence, it's more proper to say "If you would like more detail..."